rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +460,350 @@ def accepted_type(self) -> type:
     @property
     def literals(self) -> Mapping[Name, Expr]:
         return {}
+
+    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        :param check:
+        :return:
+        """
+
+        msg_as_bitstr: Bitstring = Bitstring().from_bytes(value)
+        self.assign_bitvalue(msg_as_bitstr, length)
+
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        """
+
+        current_field_name = self._next_field(model.INITIAL.name)
+        field_first_in_bitstr = 0
+        field_length = 0
+
+        while current_field_name != model.FINAL.name and (
+            field_first_in_bitstr + field_length
+        ) <= len(str(value)):
+
+            current_field = self._fields[current_field_name]
+            if isinstance(current_field.typeval, OpaqueValue) and not self._has_length(
+                current_field_name
+            ):
+
+                self._fields[current_field_name].first = self._get_first(current_field_name)
+                self.set(current_field_name, value[field_first_in_bitstr:])
+                self._fields[current_field_name].length = Number(current_field.typeval.length)
+            else:
+                assert isinstance(current_field.length, Number)
+                field_length = current_field.length.value
+
+                if field_length < 8:
+
+                    if length != 0 and length < 8:
+                        field_first_in_bitstr = field_first_in_bitstr + 8 - field_length
```

Comment:
it is used to determine the bit from which the parsing starts if the length of the message is less then one byte

Reply:
In which line?

Reply:
I found another issue with this, it needs to be changed. As you pointed out above it would be better to use an offset directly instead of computing the first bit by using the length of the message.

Reply:
Could it be that in the current implementation the offset for a message with a length of less than one byte is on the left while for longer messages the offset is on the right (of the last byte)? Shouldn't a message always start at the first bit of the first byte (even if the message is less than one byte long)?

Reply:
I was probably wrong with the offset for messages < 8 since i thought b\01 results in a byte \x01 and not x\40 as it would be if b\01 is at the beginning of the byte.